api/lib/opentelemetry/trace/tracer_factory.rb

```diff
@@ -29,6 +30,10 @@ def binary_format
       def http_text_format
         HTTP_TEXT_FORMAT
       end
+
+      def rack_http_text_format
+        RACK_HTTP_TEXT_FORMAT
+      end
```

Comment:
Given that we instantiate each of the formats once, and they’re only accessible through these methods, would it be cleaner to have a single TextFormat class and to pass the header keys to the constructor? I.e.:
```ruby
def rack_http_text_format
  @rack_http_text_format ||= TextFormat.new(tracestate_header_key: 'HTTP_TRACESTATE', traceparent_header_key: 'HTTP_TRACEPARENT')
end
```

Reply:
It would clean up a couple of things. The testing woes and having to nudge constant lookup by prefixing with `self.class::` are two things I don't particularly like.

Reply:
That would be nice! I was also [kicking around some ideas](https://github.com/open-telemetry/opentelemetry-ruby/pull/131#discussion_r342726625) on how we could make the formats easy to use and I wonder if it would also have the side effect of making the format(ters) much simpler since we wouldn't need to do the default getter/setter.

From the comment linked above - slightly changes given this conversation...

I think it would be nice to be able to do:

```ruby
# name is just an example
OpenTelemetry::Propagators::RackHTTPTextHeader.inject(span.context, request.headers)
```
versus the wiring I need to do now:
```ruby
formatter = OpenTelemetry.tracer_factory.http_text_format
# or
# formatted = OpenTelemetry::DistributedContext::Propagation::TextFormat.new

formatter.inject(span.context, request.headers) do |carrier, key, value|
  carrier[key] = value
end
```
I know #140 (default getter/setter) will help with this but I was wondering if maybe moving that to a `Propagator` would allow the formatter to stay simple and maybe we wouldn't need to expose the `format` on the factory which feels super strange to me anyway.